File: translations/codenet/gpt_commented/Go2P/codellama/Code/s376370885.py
// by syu
import sys

# Initialize a new Scanner for reading input from standard input (sys.stdin)
in = Scanner(sys.stdin, 0)

def main():
    # Read two integers n and k from the input
    n, k = in.int(), in.int()
    
    # Check if n is divisible by k
    if n % k == 0:
        # If n is divisible by k, print 0
        print(0)
    else:
        # If n is not divisible by k, print 1
        print(1)

class Scanner:
    def __init__(self, r, max):
        self.scanner = bufio.Scanner(r) # Create a new Scanner
        self.scanner.split = scanWords # Set the split function to scan words
        if max <= 0:
            max = 1048576 # Default buffer size if max is not specified
        self.scanner.buffer_size = max # Set the buffer size for the scanner

    def int(self):
        self.scanner.scan() # Scan the next token
        i, e = strconv.ParseInt(self.scanner.text, 10, 64) # Parse the token as an integer
        if e != None:
            raise e # Panic if there is an error during parsing
        return int(i) # Return the parsed integer

def scanWords(data, atEOF):
    start = 0
    # Skip leading whitespace
    for start < len(data) and isSpace(data[start]):
        start += 1
    # Find the end of the next word
    for i in range(start, len(data)):
        if isSpace(data[i]):
            return i + 1, data[start:i], None # Return the word found
    # If at EOF and there is data left, return the last word
    if atEOF and len(data) > start:
        return len(data), data[start:], None
    return start, None, None # Return the current position if no word is found

def isSpace(b):
    return b == ' ' or b == '\n' or b == '\r' or b == '\t'

def Pln(s):
    print(s) # Print the provided arguments

File: translations/codenet/gpt_commented/Go2C/deepseek/Code/s376370885.c
#include <stdio.h>
#include <stdlib.h>
#include <ctype.h>

// Custom scanner struct to read input
typedef struct {
    FILE *file;
    int bufferSize;
    char *buffer;
    size_t bufferIndex;
} Scanner;

// Initialize a new Scanner for reading input from standard input (stdin)
Scanner* newScanner(FILE *file, int bufferSize) {
    Scanner *scanner = (Scanner*)malloc(sizeof(Scanner));
    scanner->file = file;
    scanner->bufferSize = bufferSize;
    scanner->buffer = (char*)malloc(bufferSize * sizeof(char));
    scanner->bufferIndex = 0;
    return scanner;
}

// Read the next token from the input and convert it to an integer
int scanInt(Scanner *scanner) {
    int num;
    fscanf(scanner->file, "%d", &num);
    return num;
}

// Check if a character is a whitespace character
int isSpace(char c) {
    return c == ' ' || c == '\n' || c == '\r' || c == '\t';
}

// Custom split function to tokenize input
int scanWords(char *data, int *advance, char **token, int atEOF) {
    int start = *advance;
    // Skip leading whitespace
    while (start < strlen(data) && isSpace(data[start])) {
        start++;
    }
    // Find the end of the next word
    for (int i = start; i < strlen(data); i++) {
        if (isSpace(data[i])) {
            *advance = i + 1;
            *token = (char*)malloc((i - start + 1) * sizeof(char));
            strncpy(*token, data + start, i - start);
            (*token)[i - start] = '\0';
            return 1; // Return the word found
        }
    }
    // If at EOF and there is data left, return the last word
    if (atEOF && start < strlen(data)) {
        *advance = strlen(data);
        *token = (char*)malloc((strlen(data) - start + 1) * sizeof(char));
        strcpy(*token, data + start);
        return 1;
    }
    *advance = start;
    return 0; // Return the current position if no word is found
}

// Print a line to standard output
void pln(int s) {
    printf("%d\n", s);
}

int main() {
    // Initialize a new Scanner for reading input from standard input (stdin)
    Scanner *scanner = newScanner(stdin, 0);

    // Read two integers n and k from the input
    int n = scanInt(scanner);
    int k = scanInt(scanner);

    // Check if n is divisible by k
    if (n % k == 0) {
        // If n is divisible by k, print 0
        pln(0);
    } else {
        // If n is not divisible by k, print 1
        pln(1);
    }

    // Free allocated memory
    free(scanner->buffer);
    free(scanner);

    return 0;
}
